### DISCUSSION TITLE
How to run a command as root from a module?

### DISCUSSION THREAD

**Discussion by gab8192** [LSPosed/LSPosed#2308] [Q&A] [answered]
In my module, i need to run a command as root (exactly `echo "0" > /sys/fs/selinux/enforce` in order to disable selinux and therefore be able to add my custom service to ServiceManager. Because (atleast in my device) when my module tries to add a service to servicemanager, it throws SecurityException if selinux is enabled.

So, how do i run that echo command under root user? 
Keep in my mind that my module does all of its work in the system server, if that helps. I could run this command in initZygote callback, in loadPackage(framework) callback, or any later at runtime.

**Comment by yujincheng08** [comment, marked as answer]
impossible, and this is never a correct solution.

**Reply by gab8192** [reply]
Then what should i do to communicate from the UI side of my module to the system side of my module?
